mono/utils/mono-error.c

```diff
@@ -360,16 +354,13 @@ mono_error_set_type_load_name (MonoError *oerror, const char *type_name, const c
 }
 
 void
-mono_error_set_method_load (MonoError *oerror, MonoClass *klass, const char *method_name, const char *signature, const char *msg_format, ...)
+mono_error_set_method_load (MonoError *oerror, const char *message)
```

Comment:
I don't like that it's now up to each caller of `mono_error_set_method_load` to call `mono_exception_create_missing_method_message` while there's no indication that that's what you're supposed to do.

In general the thing that I liked about the old way is that you had some mandatory arguments that got formatted in a uniform way by MonoError.

Reply:
I agree with you. Ideally mono_exception_create_missing_method_message would be inlined into mono_error_set_method_load.

What I don't like is having all that metadata code in utils. What if that function lived in metadata/exceptions.c? It would call a simplified version in mono-error.c

Reply:
1. What's the rationale for mono-error being in `utils/` in the first place?

2. Suppose we rewrite another exception: `mono_error_set_type_load_class`, for example. We would presumably have `mono_exception_create_type_load_message` in `metadata/` and `mono_error_set_type_load_???` in `utils/`.  Can we make `mono_error_set_method_load` and `mono_error_set_type_load_???` into a single function?

Reply:
How about we split this so the low level code lives in utils and the high level formating code sits in metadata?

IE:

We have `mono_error_set_method_missing` in metadata/exception.c that does all formating and ends up calling `mono_error_set_specific` (or whatever) from utils/mono-error.c?

Reply:
Love it.